### DISCUSSION TITLE
Question: Query on host name (excl. domain)

### DISCUSSION THREAD

**Discussion by engstrumpan** [MichaCo/DnsClient.NET#138] [Q&A]
Hi,
As the title suggests, this is a question rather than a issue and I hope this is the way to do it.

For the record, this whole thing with DNS lookups and networking in general is not my area of expertise with little knowledge about it in general. So with that said, here's what I'm trying to accomplish.

I have a bunch of host names, i.e MY-AS-TE-ME500, that I'd like to resolve into A record IP Addresses.
Doing this command line ```nslookup MY-AS-TE-ME500``` gives me what I'm after like below (address deducted)
```
Name:    my-as-te-me500.my.domain.net
Address:  0.0.0.0
```

But, when trying out DnsClient.NET, which by the way is an awesome product and I much appreciate the effort you've put in to this, I can't query said host name and resolve the IP Address **whithout** specifying the full domain name.

**_This fails_**
```
var rs = await _lookupClient.QueryAsync("MY-AS-TE-ME500", QueryType.A);
```
**_This works_**
```
var rs = await _lookupClient.QueryAsync("MY-AS-TE-ME500.my.domain.net", QueryType.A);
```

Failing to do so, the query reaches a timeout. Clearly there is a discrepancy between ```nslookup``` and ```Dnsclient.NET```, probably for obvious reasons I don't know due to my lack of understanding of this domain.

### _So:_ 
Can this be accomplished, querying without specifying the full domain name? I've tried with specific name servers to the instantiation of the client but maybe there's other options I need to configure as well to get this to work?

edit: The reason for not wanting to specify the full domain name of the host is because I can't guarantee all server names in the list is from the same domain so can't really hard code it.

**Comment by MichaConrad** [comment]
Hi @engstrumpan 
No worries, its fine to ask questions. On Github, most repositories have a discussion section now for that though. I migrated your issue to one of those.

Regarding your question. Yeah that's a known differences between DnsClient and some other specific DNS tools.

What you are looking for is the host name suffix of your network adapter which can be appended to short names queries.
Here is what I added recently so that you can write a little bit code to do the same thing with DnsClient: https://github.com/MichaCo/DnsClient.NET/discussions/122.

I think its a pretty opinionated thing to do always do this and I'm not felling its a good idea to add this to this library.
But I'm up for suggestions how that could be added
